solidity/ecdsa/README.adoc

```diff
@@ -15,7 +17,372 @@ endif::[]
 
 toc::[]
 
-// TODO: Describe protocol
+== Overview
+
+Keep ECDSA allows creating threshold ECDSA wallets where where `n` parties share
+the power to issue digital signatures under a single public key. A threshold `t`
+is specified such that any subset of `t + 1` players can jointly sign, but any
+smaller subset cannot.
+
+`WalletRegistry` smart contract is an on-chain registry of ECDSA wallets
+controlled by off-chain network of nodes. The distributed key generation
```

Comment:
`controlled by off-chain network of nodes` -> `controlled by an off-chain network of nodes`?